### PR TITLE
Backend redesign

### PR DESCRIPTION
**Pull request by carlito1979**
Hi Jaap,

I've tested your change, and it worked, but there were a couple of other minor bugs that I've attempted to fix.
The biggest one is that the timer (which appears to be governed by the flywheel?) wasn't resetting back to zero when the rower presses the reset button on the web client.

### CODE REVIEW

File: app/engine/RowingStatistics.js [hunk 1]
```diff
@@ -65,8 +65,9 @@ function createRowingStatistics (config, session) {
     // This is the core of the finite state machine that defines all state transitions
     switch (true) {
       case (sessionStatus === 'Paused' && rower.strokeState() === 'Drive'):
-        sessionStatus = 'Rowing'
+        // the order these are done in matters
         resumeTraining()
+        sessionStatus = 'Rowing'
```
Comment (JaapvanEkris) [inline app/engine/RowingStatistics.js:68]: This feels a bit odd, as we are setting and resetting the same variable here. I think we have take a much closer look at the statemachine here, and see how it behaves in a paused scenario.

I see that we call resumeTraining() twice, once from the state machine, once as a response to an external event. I think we should add the allowResumeTraining() function, with the same content as the current resumeTraining(). Then we remove the line sessionStatus = 'WaitingForDrive' from the resumeTraining().

Comment (JaapvanEkris) [reply app/engine/RowingStatistics.js:68]: I changed your code accordingly, could you test if that works well?

**Comment by carlito1979** [comment]
Apologies Jaap, I won't push through another pull request as I clearly don't understand what I'm doing with them.

Your changes worked properly, so thanks for resolving those issues for me. I'm looking forward to seeing your engine changes make their way into the main branch in the near future!

**Comment by JaapvanEkris** [comment]
> Apologies Jaap, I won't push through another pull request as I clearly don't understand what I'm doing with them.

I know the feeling, so no problem. Changes have been added to the Backend_Redesign branch.

**Pull request closed without merging**